nixos/modules/services/web-servers/nginx/default.nix

```
@@ -362,7 +362,9 @@ let
 
         redirectListen = filter (x: !x.ssl) defaultListen;
 
-        acmeLocation = optionalString (vhost.enableACME || vhost.useACMEHost != null) ''
+        # The acme-challenge location doesn't need to be added if we are not using any automated
+        # certificate provisioning and can also be omitted when we use a certificate obtained via a DNS-01 challenge
+        acmeLocation = optionalString (vhost.enableACME || (vhost.useACMEHost != null && config.security.acme.certs.${vhost.useACMEHost}.dnsProvider == null)) ''
```

```suggestion
        acmeLocation = optionalString (vhost.enableACME || (vhost.useACMEHost != null && config.security.acme.certs.${vhost.useACMEHost}.dnsProvider != null)) ''
```

not sure if I understand well why it should not be that condition

---

If the dnsProvider is set (i.e. != null), then the certificate will use a DNS challenge and not a HTTP challenge, so there's no need for the nginx well-known/acme-challenge location

---

So we only want that block if we are not using DNS-01

---

Got it now